From cb946e2959217bcd214fb1b9a7f76225af9c42d7 Mon Sep 17 00:00:00 2001
From: "jing.zhang" <jing.zhang@amlogic.com>
Date: Tue, 15 Feb 2022 12:36:25 +0800
Subject: [PATCH] audio: dtv mediasync not free [1/1]

PD#SWPL-70805

Problem:
When only open and close did not start operation mediasync did not release

Solution:
Both creation and release use the mediasync_new variable

Verify:
rdk212
---
 audio_hal/audio_hw_dtv.c | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/audio_hal/audio_hw_dtv.c b/audio_hal/audio_hw_dtv.c
index d206a9e6..f553398b 100644
--- a/audio_hal/audio_hw_dtv.c
+++ b/audio_hal/audio_hw_dtv.c
@@ -418,16 +418,11 @@ int dtv_patch_handle_event(struct audio_hw_device *dev,int cmd, int val) {
                         Close_Dmx_Audio(demux_handle);
                         demux_handle = NULL;
                         dtv_audio_instances->demux_handle[path_id] = NULL;
-                        if ((dtvsync->mediasync_new != NULL) && (dtvsync->mediasync_new == dtvsync->mediasync)){
+                        if ((dtvsync->mediasync_new != NULL)) {
                             ALOGI("close mediasync_new:%p, mediasync:%p", dtvsync->mediasync_new, dtvsync->mediasync);
                             mediasync_wrap_destroy(dtvsync->mediasync_new);
                             dtvsync->mediasync_new = NULL;
                             dtvsync->mediasync = NULL;
-                        } 
-                        else if (dtvsync->mediasync != NULL) {
-                            ALOGI("receive close cmd, release mediasync:%p\n", dtvsync->mediasync);
-                            aml_dtvsync_release(dtvsync);
-                            dtvsync->mediasync = NULL;
                         }
                     }
                 } else {
@@ -454,7 +449,7 @@ int dtv_patch_handle_event(struct audio_hw_device *dev,int cmd, int val) {
                     patch->dtvsync = dtvsync;
                     if (dtvsync->mediasync_new != NULL) {
                         patch->dtvsync->mediasync = dtvsync->mediasync_new;
-                        dtvsync->mediasync_new = NULL;
+                        //dtvsync->mediasync_new = NULL;
                     }
                     patch->dtv_has_video = demux_info->has_video;
                     patch->demux_handle = dtv_audio_instances->demux_handle[path_id];
@@ -3334,7 +3329,7 @@ static void *audio_dtv_patch_process_threadloop_v2(void *data)
                     audio_format.format = patch->dtv_aformat;
                     mediasync_wrap_setParameter(dtvsync->mediasync_new, MEDIASYNC_KEY_AUDIOFORMAT, &audio_format);
                     patch->dtvsync->mediasync = dtvsync->mediasync_new;
-                    dtvsync->mediasync_new = NULL;
+                    //dtvsync->mediasync_new = NULL;
                 }
                 patch->dtv_first_apts_flag = 0;
                 patch->outlen_after_last_validpts = 0;
@@ -3676,7 +3671,7 @@ int create_dtv_patch_l(struct audio_hw_device *dev, audio_devices_t input,
     } else {
         patch->skip_amadec_flag = false;
     }
-
+    patch->skip_amadec_flag = true;
     if (patch->skip_amadec_flag) {
         ret = pthread_create(&(patch->audio_cmd_process_threadID), NULL,
                              audio_dtv_patch_process_threadloop_v2, patch);
-- 
2.32.0

